### PR TITLE
array and list support feature and tests

### PR DESCRIPTION
**Pull request by AliRKat**
first time touching EventTests.cs with VS Code, which caused some changes with formatting in few lines lyk

### CODE REVIEW

**Comment by AliRKat** [comment]
[[Unity] string key and array/list value should be also accepted for segmentations](https://trello.com/c/Ui8UdC3R/1871-unity-string-key-and-array-list-value-should-be-also-accepted-for-segmentations)

**Review by arifBurakDemiray — CHANGES_REQUESTED** [review]
It is all cool, thanks for your efforts. <3
Please also add unsupported data types cases, like provided list of object or map
And please also test views. Where client can provide segmentation

**Review by peterBrxwn — APPROVED** [review]

File: Assets/Tests/PlayModeTests/EventTests.cs [hunk 1]
```diff
@@ -378,7 +381,7 @@ public async void TestSegmentItemsDataTypesValidation()
 
             CountlyEventModel model = Countly.Instance.Events._eventRepo.Dequeue();
 
-            Dictionary<string, object> requireSegments = new Dictionary<string, object>{
+            Dictionary<string, object> expectedSegm = new Dictionary<string, object>{
```
Comment (arifBurakDemiray) [inline Assets/Tests/PlayModeTests/EventTests.cs:384]: maybe couple of invalid types to here

**Review by arifBurakDemiray — APPROVED** [review]

**Review by turtledreams — CHANGES_REQUESTED** [review]
Please add mixed types to the tests too, like List<object> with whatever u can imagine inside. wanna see what happens. also for naming of methods you introduced pleas go over them one more time. are they succinct, can we more generalize them etc

**Review by arifBurakDemiray — APPROVED** [review]

**Review by turtledreams — CHANGES_REQUESTED** [review]
I think the thing we talked about last time was to make the log that prints the removed key value pair a little bit better. modify that and lets merge

**Pull request merged**